### PR TITLE
feat(importCDX): enhanced sw360 CDX importer

### PR DESCRIPTION
**Pull request by sameed20**
Description:

Now, before creating a new component we will check whether this component already exist in the database or not, on the basis of VCS URL.
Similarly, we also check for the duplicate packages using the purl.
 

Closes: #2142

### CODE REVIEW

**Review by hoangnt2 — APPROVED** [review]
Changes look good

**Comment by akshitjoshii** [comment]
Tested this, working fine for different scenarios.
Capture
Capture2

File: backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentRepository.java [hunk 1]
```diff
@@ -147,6 +147,18 @@ public class ComponentRepository extends SummaryAwareRepository<Component> {
             "    }" +
             "}";
 
+    private static final String BY_VCS_LOWERCASE = "function(doc) {" +
```
Comment (afsahsyeda) [inline backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentRepository.java:150]: @sameed20 , what is the use of this view?

File: backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentRepository.java [hunk 2]
```diff
@@ -215,6 +229,13 @@ public Set<String> getComponentIdsByName(String name, boolean caseInsenstive) {
         return queryForIdsAsValue("byname", name);
     }
 
+    public Set<String> getComponentIdsByVCS(String vcs, boolean caseInsenstive){
```
Comment (afsahsyeda) [inline backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentRepository.java:232]: Why do you have an additional parameter "caseInsensitive" when it is being set to true at all the places it is being called

File: backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/PackageRepository.java [hunk 3]
```diff
@@ -48,6 +48,8 @@ public class PackageRepository extends DatabaseRepositoryCloudantClient<Package>
     private static final String BY_CREATED_ON = "function(doc) { if (doc.type == 'package') { emit(doc.createdOn, doc._id) } }";
     private static final String BY_RELEASE_ID = "function(doc) { if (doc.type == 'package') { emit(doc.releaseId, doc._id); } }";
     private static final String BY_LICENSE_IDS = "function(doc) { if (doc.type == 'package') { if (doc.licenseIds) { emit(doc.licenseIds.join(), doc._id); } else { emit('', doc._id); } } }";
+    private static final String BY_PURL = "function(doc) { if (doc.type == 'package') { emit(doc.purl.trim(), doc._id) } }";
+    private static final String BY_PURL_LOWERCASE = "function(doc) { if (doc.type == 'package') { emit(doc.purl.toLowerCase().trim(), doc._id) } }";
```
Comment (afsahsyeda) [inline backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/PackageRepository.java:52]: Please explain the use of "BY_PURL_LOWERCASE"

File: backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentRepository.java (hunk 1 above)
Comment (afsahsyeda) [reply backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/ComponentRepository.java:150]: It returns all the components irrespective of VCS

**Pull request merged**